packages/opentelemetry-sampler-aws-xray/test/remote-sampler.test.ts

```diff
@@ -0,0 +1,174 @@
+ */
+
+import * as sinon from 'sinon';
+import axios from 'axios';
+import * as nock from 'nock';
+import * as assert from 'assert';
+
+import { AWSXRayRemoteSampler } from '../src';
+
+describe('GetSamplingRules', () => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const getSamplingRulesResponseStub: any = {
+    NextToken: null,
+    SamplingRuleRecords: [
+      {
+        CreatedAt: 1.67799933e9,
+        ModifiedAt: 1.67799933e9,
+        SamplingRule: {
+          Attributes: {
+            foo: 'bar',
+            doo: 'baz',
+          },
+          FixedRate: 0.05,
+          HTTPMethod: '*',
+          Host: '*',
+          Priority: 1000,
+          ReservoirSize: 10,
+          ResourceARN: '*',
+          RuleARN: 'arn:aws:xray:us-west-2:123456789000:sampling-rule/Rule1',
+          RuleName: 'Rule1',
+          ServiceName: '*',
+          ServiceType: 'AWS::Foo::Bar',
+          URLPath: '*',
+          Version: 1,
+        },
+      },
+      {
+        CreatedAt: 0.0,
+        ModifiedAt: 1.611564245e9,
+        SamplingRule: {
+          Attributes: {},
+          FixedRate: 0.05,
+          HTTPMethod: '*',
+          Host: '*',
+          Priority: 10000,
+          ReservoirSize: 1,
+          ResourceARN: '*',
+          RuleARN: 'arn:aws:xray:us-west-2:123456789000:sampling-rule/Default',
+          RuleName: 'Default',
+          ServiceName: '*',
+          ServiceType: '*',
+          URLPath: '*',
+          Version: 1,
+        },
+      },
+      {
+        CreatedAt: 1.676038494e9,
+        ModifiedAt: 1.676038494e9,
+        SamplingRule: {
+          Attributes: {},
+          FixedRate: 0.2,
+          HTTPMethod: 'GET',
+          Host: '*',
+          Priority: 1,
+          ReservoirSize: 10,
+          ResourceARN: '*',
+          RuleARN: 'arn:aws:xray:us-west-2:123456789000:sampling-rule/Rule2',
+          RuleName: 'Rule2',
+          ServiceName: 'FooBar',
+          ServiceType: '*',
+          URLPath: '/foo/bar',
+          Version: 1,
+        },
+      },
+    ],
+  };
+
+  let clock: sinon.SinonFakeTimers;
+  let sampler: AWSXRayRemoteSampler;
+  let axiosPostSpy: sinon.SinonSpy;
+
+  const defaultEndpoint = 'http://localhost:2000';
+  const pollingInterval = 60 * 1000;
+  const config = {
+    endpoint: defaultEndpoint,
+    pollingInterval: pollingInterval,
+  };
+
+  before(() => {
+    nock('http://localhost:2000')
+      .persist()
+      .post('/GetSamplingRules')
+      .reply(200, getSamplingRulesResponseStub);
+  });
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+    axiosPostSpy = sinon.spy(axios, 'post');
+    sampler = new AWSXRayRemoteSampler(config);
+  });
+
+  afterEach(() => {
+    clock.restore();
+    axiosPostSpy.restore();
+  });
+
+  it('should throw TypeError when an invalid polling interval is passed in', async () => {
+    const configWithZeroPollingInterval = {
+      endpoint: 'http://localhost:2000',
```

Comment:
if you are testing the pollingInterval, no need to add "endpoint". Test should ideally just set up what it needs